skimage/segmentation/boundaries.py

```diff
@@ -1,41 +1,214 @@
+from __future__ import division
+
 import numpy as np
-from ..morphology import dilation, square
-from ..util import img_as_float
+from scipy import ndimage as nd
+from ..morphology import dilation, erosion, square
+from ..util import img_as_float, view_as_windows, pad
 from ..color import gray2rgb
-from .._shared.utils import deprecated
 
 
-def find_boundaries(label_img):
-    """Return bool array where boundaries between labeled regions are True."""
-    boundaries = np.zeros(label_img.shape, dtype=np.bool)
-    boundaries[1:, :] += label_img[1:, :] != label_img[:-1, :]
-    boundaries[:, 1:] += label_img[:, 1:] != label_img[:, :-1]
+def _find_boundaries_subpixel(label_img):
+    """See ``find_boundaries(..., mode='subpixel')``.
+
+    Notes
+    -----
+    This function puts in an empty row and column between each *actual*
+    row and column of the image, for a corresponding shape of $2s - 1$
+    for every image dimension of size $s$. These "interstitial" rows
+    and columns are filled as ``True`` if they separate two labels in
+    `label_img`, ``False`` otherwise.
+
+    I used ``view_as_windows`` to get the neighborhood of each pixel.
```

Comment:
"We", if you don't mind!  And let's keep it in the present.  "We use ``view_as_windows``..."